examples/generative/pixelcnn.py

```diff
@@ -0,0 +1,204 @@
+                self.mask[kernel_shape[0] // 2, kernel_shape[1] // 2, ...] = 1
+
+        # and here we apply the mask to convoltional kernal
+        outputs = self._convolution_op(inputs, self.kernel * self.mask)
+
+        if self.use_bias:
+            if self.data_format == "channels_first":
+                if self.rank == 1:
+                    # nn.bias_add does not accept a 1D input tensor.
+                    bias = array_ops.reshape(self.bias, (1, self.filters, 1))
+                    outputs += bias
+                else:
+                    outputs = nn.bias_add(outputs, self.bias, data_format="NCHW")
+            else:
+                outputs = nn.bias_add(outputs, self.bias, data_format="NHWC")
+
+        if self.activation is not None:
+            return self.activation(outputs)
+        return outputs
+
+
+# Next we build our residual block layer,
+# this is just a normal res. block but with the PixelConvLayer built in
+class ResidualBlock(keras.layers.Layer):
+    def __init__(self, filters, **kwargs):
+        super(ResidualBlock, self).__init__(**kwargs)
+        self.a = keras.layers.ReLU()
+        self.b = keras.layers.Conv2D(filters=filters, kernel_size=1, activation="relu")
+        self.c = PixelConvLayer(
+            mask_type="B",
+            filters=filters // 2,
+            kernel_size=3,
+            activation="relu",
+            padding="same",
+        )
+        self.d = keras.layers.Conv2D(filters=filters, kernel_size=1, activation="relu")
+
+    def call(self, inputs):
+        x = self.a(inputs)
+        x = self.b(x)
+        x = self.c(x)
+        x = self.d(x)
+        return keras.layers.add([inputs, x])
+
+
+"""
+# Build the model based on the original paper
+"""
+
+PixelCNN_input = keras.Input(shape=input_shape)
+x = PixelConvLayer(
+    mask_type="A", filters=128, kernel_size=7, padding="same", activation="relu"
+)(PixelCNN_input)
+
+for _ in range(n_residual_blocks):
+    x = ResidualBlock(filters=128)(x)
+x = layers.ReLU()(x)
+
+for _ in range(2):
+    x = PixelConvLayer(
+        mask_type="B",
+        filters=128,
+        kernel_size=1,
+        strides=1,
+        activation="relu",
+        padding="valid",
+    )(x)
+
+x = keras.layers.Conv2D(
+    filters=1, kernel_size=1, strides=1, activation="sigmoid", padding="valid"
+)(x)
+
+PixelCNN = keras.Model(PixelCNN_input, x)
+adam = keras.optimizers.Adam(learning_rate=0.0001)
+PixelCNN.compile(optimizer=adam, loss="binary_crossentropy")
+
+PixelCNN.summary()
+PixelCNN.fit(
+    x=data.astype("float32"),
+    y=data.astype("float32"),
+    batch_size=64,
+    epochs=50,
+    validation_split=0.1,
+)
+
+"""
+# Demo
+
+The PixelCNN cannot create the full image at once and must instead create each pixel in
+order, append the next created pixel to current image, and feed the image back into the
+model to repeat the process.
+"""
+
+from IPython.display import Image, display
+from tqdm import tqdm
+from scipy.stats import bernoulli
```

Comment:
You may be able to use TF functionality instead of importing scipy, which would reduce the dependency footprint